src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -2116,6 +2116,9 @@ private static SyntaxNode WithParameterList(SyntaxNode declaration, BaseParamete
                             .WithLeadingTrivia(lambda.GetLeadingTrivia())
                             .WithTrailingTrivia(lambda.GetTrailingTrivia());
                     }
+                case SyntaxKind.RecordDeclaration:
+                case SyntaxKind.RecordStructDeclaration:
+                    return ((RecordDeclarationSyntax)declaration).WithParameterList((ParameterListSyntax)list);
```

Comment:
Does someone think it's worth adding a validation in debug mode that uses reflection to make sure new nodes are handled? (This doesn't apply to this method only - there are many that usually get affected)